adapters/repos/db/migrator.go

```diff
@@ -253,6 +253,63 @@ func (m *Migrator) RecalculateVectorDimensions(ctx context.Context) error {
 	return nil
 }
 
+func (m *Migrator) RecountProperties(ctx context.Context) error {
+	count := 0
+	m.logger.
+		WithField("action", "recount").
+		Info("Recounting properties, this may take a while")
+
+	clearedShard := map[string]bool{}
+
+	// Iterate over all indexes
+	for _, index := range m.db.indices {
```

Comment:
you should lock the indices during recount so no classes are added/deleted

Reply:
I think better be safe than sorry with this. If nothing else has started yet it doesn't hurt to lock